[TensorExpr] Allow for 'keepdim' argument in aten::mean in NNC's external call.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #66522
* #66847
* #66515
* #66514
* #69055
* #68857
* #68856
* #68855
* #68854
* #68757
* __->__ #68756
* #68755

That fixes some warnings in our tests.

Differential Revision: [D32600952](https://our.internmc.facebook.com/intern/diff/D32600952)

Commit:
[TensorExpr] Allow for 'keepdim' argument in aten::mean in NNC's external call.

That fixes some warnings in our tests.

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32600952).

navahgar commented on torch/csrc/jit/tensorexpr/external_functions.cpp:
```diff
@@ -467,12 +467,13 @@ void nnc_aten_mean(
+  bool keepdim = (bool)extra_args[args_num - 1];
   if (args_num > 0) {
-    memcpy(mean_dims.data(), extra_args, sizeof(int64_t) * args_num);
+    memcpy(mean_dims.data(), extra_args, sizeof(int64_t) * (args_num - 1));
```
This allows 0 sized `memcpy`. Can that happen here? If not, can we guard against it?

ZolotukhinM replied:
It shouldnt be possible (but also we can't really check it since we don't have error reporting mechanism yet).

navahgar reviewed (approved):
Mostly LGTM.

Commit:
Update on "[TensorExpr] Allow for 'keepdim' argument in aten::mean in NNC's external call."


That fixes some warnings in our tests.

Differential Revision: [D32600952](https://our.internmc.facebook.com/intern/diff/D32600952)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32600952).

Commit:
Update on "[TensorExpr] Allow for 'keepdim' argument in aten::mean in NNC's external call."


That fixes some warnings in our tests.

Differential Revision: [D32600952](https://our.internmc.facebook.com/intern/diff/D32600952)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32600952).

Commit:
Update on "[TensorExpr] Allow for 'keepdim' argument in aten::mean in NNC's external call."


That fixes some warnings in our tests.

Differential Revision: [D32600952](https://our.internmc.facebook.com/intern/diff/D32600952)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32600952).
